static/js/stream_list.js

```diff
@@ -124,30 +125,43 @@ export function build_stream_list(force_rerender) {
     function add_sidebar_li(stream_id) {
         const sidebar_row = stream_sidebar.get_row(stream_id);
         sidebar_row.update_whether_active();
+        // sidebar_row.update_whether_muted_active();
         elems.push(sidebar_row.get_li());
     }
 
     topic_list.clear();
     parent.empty();
 
+    const any_pinned_streams = stream_groups.pinned_streams.length > 0;
+    const any_normal_streams = stream_groups.normal_streams.length > 0;
+    const any_dormant_streams = stream_groups.dormant_streams.length > 0;
+
+    if (any_pinned_streams) {
+        elems.push('<div class="streams_subheader"><span>Pinned</span></div>');
+    }
+
     for (const stream_id of stream_groups.pinned_streams) {
         add_sidebar_li(stream_id);
     }
 
-    const any_pinned_streams = stream_groups.pinned_streams.length > 0;
-    const any_normal_streams = stream_groups.normal_streams.length > 0;
-    const any_dormant_streams = stream_groups.dormant_streams.length > 0;
+    for (const stream_id of stream_groups.muted_pinned_streams) {
+        add_sidebar_li(stream_id);
+    }
 
-    if (any_pinned_streams && (any_normal_streams || any_dormant_streams)) {
-        elems.push('<hr class="stream-split">');
+    if (any_normal_streams) {
+        elems.push('<div class="streams_subheader"><span>Active</span></div>');
```

Comment:
I think we should have a template for this with Active/Inactive/Pinned as template variable. Injecting raw html like this appears a bit unconventional.

Reply:
@m-e-l-u-h-a-n as template variable like string template declared at the start of the file or a different .hbs file?

Reply:
I meant creating a different `.hbs` file and passing Active/Inactive/Pinned as template variable to this file. It would be easy to make changes later if any design/html change is needed for this. May be you can also do this in it's own separate commit.

Reply:
@m-e-l-u-h-a-n In commit e556623, I have implemented the .hbs version. But while writing the tests, it requires the mock to be present in every tests of `stream_list.js`, event if it's not required.

Reply:
That's because those test call `stream_list.build_stream_list();` that is where these elements are rendered. So you need them to have as a mock in the test.

Also we can wait for Tim/Alya's call on this if you are not sure about making this change.

Reply:
We generally avoid writing any HTML outside HTML templates where possible, so yeah a template (or section within the template for the stream list) sounds correct.